### PR TITLE
[INLONG-9112][Agent] Add task and instance profile

### LINKED ISSUE #9112: [Improve][Agent] Add task and instance profile
### Description

Add task and instance profile to serialize task and instance config

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
### Prepare a Pull Request
[INLONG-9112][Agent] Add task and instance profile

- Fixes #9112 

### Motivation

Add task and instance profile to serialize task and instance config

### Modifications

Add task and instance profile, and add taskProfileDto to translate manager configure into task profile

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+        task.setChannel(DEFAULT_CHANNEL);
+        task.setIp(dataConfig.getIp());
+        task.setOp(dataConfig.getOp());
+        task.setDeliveryTime(dataConfig.getDeliveryTime());
+        task.setUuid(dataConfig.getUuid());
+        task.setVersion(dataConfig.getVersion());
+        task.setState(dataConfig.getState());
+
+        // set sink type
+        if (dataConfig.getDataReportType() == 0) {
```
Comment (EMsnap) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java:414]: 0 and 1 can be constant with meaningful name , the best way is making type a enum

Comment (justinwwhuang) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java:414]: ok

**Review by doleyzi — APPROVED** [review]

**Comment by dockerzhang** [comment]
@justinwwhuang please rebase from the master

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+import lombok.Data;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.inlong.agent.constant.FetcherConstants.AGENT_MANAGER_VIP_HTTP_HOST;
+import static org.apache.inlong.agent.constant.FetcherConstants.AGENT_MANAGER_VIP_HTTP_PORT;
+import static org.apache.inlong.agent.constant.TaskConstants.SYNC_SEND_OPEN;
+import static org.apache.inlong.common.enums.DataReportTypeEnum.NORMAL_SEND_TO_DATAPROXY;
+
+@Data
+public class TaskProfileDto {
```
Comment (EMsnap) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java:38]: A DTO class means data transfer object, however here you provide 9 constructors for different jobs which is the function of a factory class. Maybe those constructor should be in a factory.

Comment (justinwwhuang) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/pojo/TaskProfileDto.java:38]: ok, i will fix it in other issue

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**